trieste/models/keras/models.py

```diff
@@ -0,0 +1,293 @@
+    can be supplied, as long as it has a Gaussian distribution as a final layer and follows the
+    :class:`~trieste.models.keras.KerasEnsembleNetwork` interface.
+
+    Note that setting up the model with dictionary configs is not currently possible.
+
+    When using the model we recommend to set `tf.keras.backend.set_floatx(tf.float64)` for
+    alignment with the Trieste toolbox.
+    """
+
+    def __init__(
+        self,
+        model: KerasEnsemble,
+        optimizer: Optional[KerasOptimizer] = None,
+        bootstrap: bool = False,
+    ) -> None:
+        """
+        :param model: A Keras ensemble model with probabilistic networks as ensemble members. The
+            model has to be built but not compiled.
+        :param optimizer: The optimizer wrapper with necessary specifications for compiling and
+            training the model. Defaults to :class:`~trieste.models.optimizer.KerasOptimizer` with
+            :class:`~tf.optimizers.Adam` optimizer, negative log likelihood loss and a dictionary
+            of default arguments for Keras `fit` method: 100 epochs, batch size 100, and verbose 0.
+            See https://keras.io/api/models/model_training_apis/#fit-method for a list of possible
+            arguments.
+        :param bootstrap: Sample with replacement data for training each network in the ensemble.
+            By default set to `False`.
+        :raise ValueError: If ``model`` is not an instance of
+            :class:`~trieste.models.keras.KerasEnsemble` or ensemble has less than two base
+            learners (networks).
+        """
+
+        if not isinstance(model, KerasEnsemble):
+            raise ValueError(
+                f"In DeepEnsemble model must be an instance of a KerasEnsemble, "
+                f"received {type(model)} instead."
+            )
+
+        if optimizer is None:
+            optimizer = KerasOptimizer(tf.keras.optimizers.Adam())
+        self._optimizer = optimizer
+
+        if not isinstance(optimizer.optimizer, tf.keras.optimizers.Optimizer):
+            raise ValueError(
+                f"Optimizer for `DeepEnsemble` model must be an instance of a "
+                f"`tf.keras.optimizers.Optimizer`, "
+                f"received {type(optimizer.optimizer)} instead."
+            )
+
+        super().__init__(optimizer)
+
+        if not self.optimizer.fit_args:
+            self.optimizer.fit_args: Optional[Dict[str, Any]] = {
+                "verbose": 0,
+                "epochs": 100,
+                "batch_size": 100,
+            }
+
+        if self.optimizer.loss is None:
+            self.optimizer.loss = negative_log_likelihood
+
+        model.model.compile(
+            self.optimizer.optimizer,
+            loss=[self.optimizer.loss] * model.ensemble_size,
+            metrics=[self.optimizer.metrics] * model.ensemble_size,
+        )
+
+        self._model = model
+        self._bootstrap = bootstrap
+
+        if self.ensemble_size < 2:
```

Comment:
Since you're still initialising, it might be clearer to move this test up and not use the property?

```suggestion
        if model.ensemble_size < 2:
```